#(pf (version 1.0))
(qrontab
 (param(key globalparam1)(value foobar))

 # tasks
 (taskgroup(id appli.batch)
 )
 (task (id batch3)(taskgroup appli.batch)(mean exec)
  (command /bin/false)
  (target localhost)
  (trigger(cron 2/10 * * * * *))
  (resource(kind memory)(quantity 103))
 )
 (task (id batch4)(taskgroup appli.batch)(mean ssh)(target localhost)
  (command BARFOO=14 /tmp/foo %!yyyy)
  (trigger(cron 1/10 * * * * *))
  (resource(kind memory)(quantity 104))
 )

 # hosts
 (host(id localhost)(hostname localhost)(resource(kind memory)(quantity 128)))

 # logs
 (log(level debug)(file "/tmp/qron-debug-${%Y}-${%m}-${%d}.log"))
 (log(level info)(file "/tmp/qron-${%Y}-${%m}-${%d}.log")) # default: repoen/10'

 # alerts
 (alerts
  (rule(match task.failure.appli.batch.*) # task.failure.%!fqtn
        (mail(address gb@localhost))
   )
   (rule(match task.failure.appli.batch.*) # task.failure.%!fqtn
        (udp(address 127.0.0.1:4242)
            (message task failure: %!alertid)
            (cancelmessage task back to success: %!alertid))
        (mail(address nobody@localhost)) # this is an error 
        (stop) # do not process next rules if this one matches
   )
   (rule(match resource.exhausted.**) # resource.exhausted.%!kind.%!target
        (mail(address gb@localhost))
        #(nocancelnotify) # do not send messages when alerts are canceled
   )
   (rule(log(address error)) # match every alert (but those stopped before)
        (udp(address localhost:4243))
   )
   (param(key mail.mindelaybetweenmails)(value 60))
   (param(key canceldelay)(value 90))
 )
)
